### PR TITLE
add readme to obstacle collision checker

### PR DESCRIPTION
**Pull request by taikitanaka3**
## Description

add readme to obstacle collision checker

## Review Procedure

add readme to obstacle collision checker

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [pull request guidelines][pull-request-guidelines]
- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: control/trajectory_follower/obstacle_collision_checker/README.md [hunk 1]
```diff
@@ -1 +1,87 @@
 # obstacle_collision_checker
+
+## Purpose
+
+`obstacle_collision_checker` is a module to check obstacle collision for predicted path and publish `ERROR` diag if collision is found.
```
Comment (kmiya) [inline control/trajectory_follower/obstacle_collision_checker/README.md:5]: IMO: diag はあまり一般的な略語ではないようなので、略さずに diagnostic message (？)等とするか、 multi_object_tracker や imu_corrector の README のように `ERROR` diag → diagnostic errors とするのはいかがでしょうか。実際の仕様と違ったらすみません。

File: control/trajectory_follower/obstacle_collision_checker/README.md [hunk 2]
```diff
@@ -1 +1,87 @@
...
+@enduml
+```
+
+### Algorithms
+
+### Check data
+
+Check that `obstacle_collision_checker` receives no ground pointcloud, predicted_trajectory, reference trajectory, and current velocity data.
+
+### Diagnostic Update
```
Comment (kmiya) [inline control/trajectory_follower/obstacle_collision_checker/README.md:53]:
```suggestion
### Diagnostic update
```
Nit: 他と合わせて2単語目以降は小文字でしょうか？

File: control/trajectory_follower/obstacle_collision_checker/README.md [hunk 3]
```diff
@@ -1 +1,87 @@
...
+
+If any collision is found on predicted path, this module sets `ERRROR` level as diagnostic status else sets `OK`.
```
Comment (kmiya) [inline control/trajectory_follower/obstacle_collision_checker/README.md:55]:
```suggestion
If any collision is found on predicted path, this module sets `ERROR` level as diagnostic status else sets `OK`.
```
Nit: ERRROR → ERROR

File: control/trajectory_follower/obstacle_collision_checker/README.md [hunk 4]
```diff
@@ -1 +1,87 @@
...
+| Name             | Type                                   | Description              |
+| ---------------- | -------------------------------------- | ------------------------ |
+| `~/debug/marker` | `visualization_msgs::msg::MarkerArray` | Marker for visualization |
+
+## Parameters
+
+| Name                | Type     | Description                                      | Default value |
+| :------------------ | :------- | :----------------------------------------------- | :------------ |
+| `delay_time`        | `double` | delay time of vehicle [s]                        | 0.3           |
+| `footprint_margin`  | `double` | foot print margin [m]]                           | 0.0           |
```
Comment (kmiya) [inline control/trajectory_follower/obstacle_collision_checker/README.md:80]:
```suggestion
| `footprint_margin`  | `double` | foot print margin [m]                            | 0.0           |
```
Nit: 不要な`]`がありました。

File: control/trajectory_follower/obstacle_collision_checker/README.md [hunk 5]
```diff
@@ -1 +1,87 @@
...
+
+| Name             | Type                                   | Description              |
+| ---------------- | -------------------------------------- | ------------------------ |
+| `~/debug/marker` | `visualization_msgs::msg::MarkerArray` | Marker for visualization |
+
+## Parameters
+
+| Name                | Type     | Description                                      | Default value |
+| :------------------ | :------- | :----------------------------------------------- | :------------ |
+| `delay_time`        | `double` | delay time of vehicle [s]                        | 0.3           |
```
Comment (kmiya) [inline control/trajectory_follower/obstacle_collision_checker/README.md:79]: IMO: 他の表と表記を合わせ、 Description の列の1文字目は大文字にするのはいかがでしょうか。他の行も同様です。

Comment (taikitanaka3) [reply control/trajectory_follower/obstacle_collision_checker/README.md]: 773cbef

File: control/trajectory_follower/obstacle_collision_checker/README.md (hunk 1 above)
Comment (taikitanaka3) [reply control/trajectory_follower/obstacle_collision_checker/README.md]: publish diagnostic errorsとしました。
9c1a1c4

File: control/trajectory_follower/obstacle_collision_checker/README.md [hunk 6]
```diff
@@ -1 +1,87 @@
 # obstacle_collision_checker
+
+## Purpose
+
+`obstacle_collision_checker` is a module to check obstacle collision for predicted path and publish diagnostic errors if collision is found.
```
Comment (tkimura4) [inline control/trajectory_follower/obstacle_collision_checker/README.md:5]:
```suggestion
`obstacle_collision_checker` is a module to check obstacle collision for predicted trajectory and publish diagnostic errors if collision is found.
```

**Review by tkimura4 — APPROVED** [review]
LGTM

**Pull request merged**